read-all-devices/src/main.rs

```diff
@@ -1,53 +1,208 @@
-use blurz::{BluetoothDevice, BluetoothEvent, BluetoothSession};
+use bluez_generated::bluetooth_event::BluetoothEvent;
+use bluez_generated::generated::adapter1::OrgBluezAdapter1;
+use bluez_generated::generated::device1::OrgBluezDevice1;
+use bluez_generated::generated::gattcharacteristic1::OrgBluezGattCharacteristic1;
+use dbus::arg::RefArg;
+use dbus::nonblock::stdintf::org_freedesktop_dbus::ObjectManager;
+use dbus::nonblock::SyncConnection;
+use futures::FutureExt;
+use futures::StreamExt;
+use itertools::Itertools;
 use mijia::{
-    connect_sensors, decode_value, find_sensors, hashmap_from_file, print_sensors, scan,
-    start_notify_sensors, SERVICE_CHARACTERISTIC_PATH,
+    decode_value, CONNECTION_INTERVAL_500_MS, CONNECTION_INTERVAL_CHARACTERISTIC_PATH,
+    MIJIA_SERVICE_DATA_UUID, SERVICE_CHARACTERISTIC_PATH,
 };
-use std::thread;
+use std::collections::HashMap;
+use std::error::Error;
+use std::sync::Arc;
 use std::time::Duration;
+use std::time::Instant;
 
-mod explore_device;
-
-const SENSOR_NAMES_FILENAME: &str = "sensor_names.conf";
-
-fn main() {
-    let sensor_names = hashmap_from_file(SENSOR_NAMES_FILENAME).unwrap();
-    let bt_session = &BluetoothSession::create_session(None).unwrap();
-    let device_list = scan(&bt_session).unwrap();
-    let sensors = find_sensors(&bt_session, &device_list);
-    println!();
-    print_sensors(&sensors, &sensor_names);
-    let connected_sensors = connect_sensors(&sensors);
-    print_sensors(&connected_sensors, &sensor_names);
-
-    // We need to wait a bit after calling connect to safely
-    // get the gatt services
-    thread::sleep(Duration::from_millis(5000));
-    for device in &connected_sensors {
-        explore_device::explore_gatt_profile(bt_session, &device);
+#[tokio::main(core_threads = 3)]
```

Comment:
- [ ] we can get away with letting tokio decide how many cores it wants now, because nothing is blocking for more than a fraction of a second.